code/modules/mob/living/carbon/xenomorph/castes/queen/abilities_queen.dm

```diff
@@ -332,10 +332,29 @@
 /datum/action/xeno_action/set_xeno_lead/proc/select_xeno_leader(mob/living/carbon/xenomorph/selected_xeno, feedback = TRUE)
 	var/mob/living/carbon/xenomorph/queen/xeno_ruler = owner
 	if(QDELETED(selected_xeno))
-		var/list/possible_xenos = xeno_ruler.hive.get_leaderable_xenos()
-
-		selected_xeno = tgui_input_list(xeno_ruler, "Target", "Watch which xenomorph?", possible_xenos)
-		if(QDELETED(selected_xeno) || selected_xeno.stat == DEAD || is_centcom_level(selected_xeno.z))
+		var/list/possible_xeno_names = xeno_ruler.hive.get_leaderable_xenos()
+
+		var/selected_name = tgui_input_list(xeno_ruler, "Target", "Select which xenomorph?", possible_xeno_names)
+
+		//Cleaning name of prefix and suffix. Actual xeno name will be in group[1].
+		var/regex/name_finder = regex(@"(?:\(L\) |^)(.*?)(?:(?<=\)) \(SSD\)|$)")
+		name_finder.Find(selected_name)
```

Comment:
this does nothing?

Reply:
This regex is removing the (L) and (SSD) added on to the listing.

Reply:
This regex removes the (L) and (SSD) added on in the list of leaders.